### DISCUSSION TITLE
Default use (examples of) the partials folders?

### DISCUSSION THREAD

**Discussion by guyinpv** [TukuToi/better-wp-plugin-boilerplate#20] [General]
I've been studying the plugin and the only two things that seem to have no connected use, are the "partials" folders. There is one in the admin folder and one in the public folder. There are two php files `....admin-display.php` and `...public-display.php`. These aren't referenced anywhere in the plugin so it's not really clear how to make use of the "partials" folders in the first place.

It's obvious these are used for some kind of templating, but since they are named "partials", I have to assume there should be some other templates that aren't partials? Should there be a `/templates` folder along with partials?

I think either remove partials altogether, or make the templating abilities a little more useful out of the box so that we can do something with them right away.

I think having these two partial templates sitting there with no connection to anything, isn't that helpful. Either build them out so that they are used even in a basic way (templating front and back), or just remove them entirely. Otherwise as it is, it's just kind of confusing since there is no boilerplate for how to load them or anything.

Thanks!

**Comment by smileBeda** [comment]
Hey @guyinpv - I am sorry the late reply here.
I do not get notifications for these discussions for some reason. I will have to explore why and fix that.

For now, let me reply (too late) to your comment.
> I've been studying the plugin and the only two things that seem to have no connected use, are the "partials" folders. There is one in the admin folder and one in the public folder. There are two php files ....admin-display.php and ...public-display.php. These aren't referenced anywhere in the plugin so it's not really clear how to make use of the "partials" folders in the first place.

Those files are intended to be "mostly html" _views_ (if we speak in MVP terms)
You craft your logic in the class(es) and at some point you want to display a bunch of things.
Like let's say a Form, or a Plugin Admin page with lots of nice HTML and few caller functions that display some data from your plugin

That is what those files are for
You can see a good usage example here: https://github.com/TukuToi/tukutoi-plugin-generator/blob/main/public/partials/tkt-plugin-generator-public-display.php
That file mostly is HTML, it represents a Form that gets submitted by a user. The plugin then reads the stuff in the class here https://github.com/TukuToi/tukutoi-plugin-generator/blob/main/public/class-tkt-plugin-generator-public.php

Or, another example is https://github.com/TukuToi/tukutoi-template-builder/tree/main/admin/partials 
You will see _many_ partials there. Each represents a small chunk of HTML + PHP to _display_ a user interface.
You can then `include` those files like so https://github.com/TukuToi/tukutoi-template-builder/blob/7f2b182d5f6783c91564d54dbd8bf6dfb840181a/admin/class-tkt-template-builder-admin.php#L392
Including a file in the PHP flow will basically make it "part" of that code.
The other files in that same folder are included in this case here https://github.com/TukuToi/tukutoi-template-builder/blob/7f2b182d5f6783c91564d54dbd8bf6dfb840181a/admin/class-tkt-template-builder-admin.php#L804, and only if another plugin is active :)
Again they just display some HTML GUI with a bit of PHP

The core idea of this is to allow you to separate concerns (display, logic)

The name `partials` is a bit arbitrary and bit reflected. It _is_ a `partial` in the sense of (like in my last example above) it can describe a _part_ of a bigger, later bootstrapped thing, or, it is a part of a bigger code that - in this partial - becomes visible.
You could also name it templates, of course, or simply "html"

The cannot be "used" by the boilerplate because the boilerplate does not know what you want to do with it.
You can, if you do not use them, delete the entire folder(s)
If your plugin has a backend (or front end) _view_ then you can use them. Many plugins don't

Hope it makes sense

**Comment by guyinpv** [comment]
Thanks again, it makes sense how you use them. 
I think in my brain I was expecting some sort of templating engine to be used, like Twig or something, where there is a "view" engine that grabs template files as needed and passes data to them and so forth, rather than simply including php files.

In one of my small plugins, for things like shortcodes or email body content, it's so short that I just inlined the HTML strings rather than include separate template files.

I wonder if a small templating engine wouldn't be a bad thing for the boilerplate? Maybe not Twig if that's too heafty, but some simple means to call upon templates with inheritance ability, and pass data to them. Just a thought!

**Reply by smileBeda** [reply]
That’s actually what the file allows you to do. Including it, automatically makes it part of the code where you include it, and thus any “$this” or other variable available to the function that includes the file also is available in the file

twig would be overkill I think specially since not everyone might want or need such template engine.

but it’s of course perfectly possible to include it too. However that’d be plugin specific Development that I don’t see part of a boilerplate 

but if someone would pr such enhancements I’d certainly consider including them!
